submission/races.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package races
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"reflect"
+	"sync"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/ctpolicy"
+)
+
+const (
+	// URLProtocol keeps protocol substring for Log URLs.
+	URLProtocol = "https://"
+
+	// Sleeptim is duration between parallel batch call and subsequent requests to Logs within group.
+	SleepTime = time.Second
+)
+
+type guardedLogs struct {
```

Comment:
There's nothing log-specific about this, so the naming is misleading -- it describes how you happen to want to use it, rather than what it does.  AFAICT it's really a `safeStringSet` or somesuch?